stackslib/src/chainstate/nakamoto/coordinator/tests.rs

```diff
@@ -486,6 +973,691 @@ fn test_simple_nakamoto_coordinator_10_tenures_10_blocks() {
+
+        let stx_balance = clarity_instance
+            .read_only_connection(&block_id, &chainstate_tx, &sort_db_tx)
+            .with_clarity_db_readonly(|db| db.get_account_stx_balance(&miner.clone().into()));
+
+        // only count matured rewards (last 3 blocks are not mature)
+        let block_fee = if i > 3 {
+            fee_counts[i.saturating_sub(4)]
+        } else {
+            0
+        };
+        let expected_total_tx_fees = fees_so_far + block_fee;
+        let expected_total_coinbase = expected_coinbase_rewards;
+        fees_so_far += block_fee;
+
+        if i == 0 {
+            // first tenure awards the last of the initial mining bonus
+            expected_coinbase_rewards += (1000 + 2600) * 1000000;
+        } else {
+            // subsequent tenures award normal coinbases
+            expected_coinbase_rewards += 1000 * 1000000;
+        }
+
+        eprintln!(
+            "Checking block #{} ({},{}): {} =?= {} + {}",
+            i,
+            &ch,
+            &sn.block_height,
+            stx_balance.amount_unlocked(),
+            expected_total_coinbase,
+            expected_total_tx_fees
+        );
+        assert_eq!(
+            stx_balance.amount_unlocked(),
+            expected_total_coinbase + expected_total_tx_fees
+        );
+    }
+
+    let tip = {
+        let chainstate = &mut peer.stacks_node.as_mut().unwrap().chainstate;
+        let sort_db = peer.sortdb.as_mut().unwrap();
+        NakamotoChainState::get_canonical_block_header(chainstate.db(), sort_db)
+            .unwrap()
+            .unwrap()
+    };
+
+    assert_eq!(
+        tip.anchored_header
+            .as_stacks_nakamoto()
+            .unwrap()
+            .chain_length,
+        (11 + total_blocks) as u64
+    );
+    assert_eq!(
+        tip.anchored_header.as_stacks_nakamoto().unwrap(),
+        &rc_blocks.last().unwrap().last().unwrap().header
+    );
+
+    // verify that matured miner records were in place
+    let mut matured_rewards = vec![];
+    {
+        let chainstate = &mut peer.stacks_node.as_mut().unwrap().chainstate;
+        let sort_db = peer.sortdb.as_mut().unwrap();
+        let (mut chainstate_tx, _) = chainstate.chainstate_tx_begin().unwrap();
+        for i in 0..24 {
+            let matured_reward_opt = NakamotoChainState::get_matured_miner_reward_schedules(
+                &mut chainstate_tx,
+                &tip.index_block_hash(),
+                i,
+            )
+            .unwrap();
+            matured_rewards.push(matured_reward_opt);
+        }
+    }
+    for (i, matured_reward_opt) in matured_rewards[4..].into_iter().enumerate() {
+        let matured_reward = (*matured_reward_opt).clone().unwrap();
+        debug!("{}: {:?}", i, &matured_reward);
+
+        if i < 10 {
+            assert_eq!(matured_reward.parent_miner.coinbase, 3600_000_000);
+        } else {
+            assert_eq!(matured_reward.parent_miner.coinbase, 1000_000_000);
+        }
+
+        if i < 11 {
```

Comment:
Super nit especially since in a test, ut this would be clearer to me with a match on a std::cmp
Like so 
```
    match i.cmp(&11) {
        Ordering::Less => {...},
        Ordering::Greater => {...},
        Ordering::Equal => {...},
    }
    ```